### PR TITLE
Update Changelog template

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

As title. Leaving the below sections unaltered to illustrate.

This workflow for selecting a category is a bit easier / more obvious.

## Changelog

Insert changelog entry or "NOCHANGELOG" here.

- [ ] `NEW` _(added new feature or capability)_
- [ ] `BUGFIX` _(fixed a bug)_
- [ ] `DOCS` _(added or updated documentation)_

### CODE REVIEW

**Comment by OwenKephart** [comment]
* **#24214** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24214?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @OwenKephart and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by smackesey — APPROVED** [review]
🙏

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -2,6 +2,18 @@
 
 ## How I Tested These Changes
 
-## Changelog [New | Bug | Docs]
+## Changelog
 
-> Replace this message with a changelog entry, or `NOCHANGELOG`
+> Replace this section with a changelog entry, or `NOCHANGELOG` if no entry is required (e.g. an internal refactor).
```
Comment (sryza) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: I generally like how concise our current template is. We had a template before where I would need to delete a bunch of lines every time I posted, and I was very happy when we switched to the current one. Not 100% confident what the right balance, but would generally bias a little more towards brevity than what's here.  Also, for the educational content that we do keep here, better to use comments than indented quotes.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -2,6 +2,18 @@
...
+>
+> If you are editing an integration library, add `[dagster-<integration>]` to the start of your entry:
+>
+> `[dagster-dbt] Fixed ...`
+
+## Category
+
+> Select a category for this PR
+
+- [ ] `NEW` _(added new feature or capability)_
```
Comment (sryza) [inline .github/PULL_REQUEST_TEMPLATE.md:17]: Awesome, a lot smoother

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (smackesey) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Agree. How about:

```
Insert changelog entry or "NOCHANGELOG" here.
```

Also I think we could probably nuke the whole "Category" section and instead just stick the category boxes right under the "## Changelog" header?

Comment (sryza) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Other idea to consider is just letting people delete the "## Changelog" section to indicate no changelog?

Comment (OwenKephart) [reply .github/PULL_REQUEST_TEMPLATE.md]: @sryza , I think deleting the section == no changelog is likely good idea in the medium term, but I've been trying to guard against old PRs without this section slipping through the cracks. So maybe in a couple weeks we can change the flow again

**Pull request merged**